kotlin-android/app/src/main/java/com/mongodb/tasktracker/TaskActivity.kt

```diff
@@ -0,0 +1,150 @@
+    private lateinit var fab: FloatingActionButton
+
+    override fun onStart() {
+        super.onStart()
+        handleLogin()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        user.run {
+            realm.close()
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_task)
+        // default instance uses the configuration created in the login activity
+        realm = Realm.getDefaultInstance()
+        recyclerView = findViewById(R.id.task_list)
+        fab = findViewById(R.id.floating_action_button)
+
+        // create a dialog to enter a task name when the floating action button is clicked
+        fab.setOnClickListener {
+            val input = EditText(this)
+            val dialogBuilder = AlertDialog.Builder(this)
+            dialogBuilder.setMessage("Enter task name:")
+                    .setCancelable(true)
+                    .setPositiveButton("Create") { dialog, _ -> run {
+                        dialog.dismiss()
+                        val task = Task(input.text.toString())
+                        // all realm writes need to occur inside of a transaction
+                        realm.executeTransactionAsync { realm ->
+                            realm.insert(task)
+                        }
+                    }
+                    }
+                    .setNegativeButton("Cancel") { dialog, _ -> dialog.cancel()
+                    }
+
+            val dialog = dialogBuilder.create()
+            dialog.setView(input)
+            dialog.setTitle("Create New Task")
+            dialog.show()
+        }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        recyclerView.adapter = null
+        // if a user hasn't logged out when we close the realm, still need to explicitly close
+        realm.close()
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.activity_task_menu, menu)
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.action_logout -> {
+                user?.logOutAsync {
+                    if (it.isSuccess) {
+                        // always close the realm when finished interacting to free up resources
+                        realm.close()
+                        user = null
+                        handleLogin()
+                        Log.v(TAG(), "user logged out")
+                    } else {
+                        RealmLog.error(it.error.toString())
+                        Log.e(TAG(), "log out failed! Error: ${it.error}")
+                    }
+                }
+                true
+            }
+            else -> {
+                super.onOptionsItemSelected(item)
+            }
+        }
+    }
+
+
+    private fun handleLogin() {
```

Comment:
It is a slightly confusing name. It is really validating that the user is logged in and opening the Realm. The fact that it has to be described awkwardly gives you a hint that this method is doing too much. I would just leave most of this in onStart() and when you log out, just use the `startActivity(Intent(this, LoginActivity::class.java))` line again so it's perfectly clear what you're doing.